packages/flutter/lib/src/material/material_state.dart

```diff
@@ -651,7 +651,9 @@ abstract class MaterialStateProperty<T> {
 
   /// Convenience method for creating a [MaterialStateProperty] that resolves
   /// to a single value for all states.
-  static MaterialStateProperty<T> all<T>(T value) => _MaterialStatePropertyAll<T>(value);
+  ///
+  /// If you need a const value, use [MaterialStatePropertyAll] directly.
+  static MaterialStateProperty<T> all<T>(T value) => MaterialStatePropertyAll<T>(value);
```

Comment:
Add a TODO to deprecate this and link it to the dartfix issue that needs to be resolved to do so?